### PR TITLE
[Feature][Quantization] Support compressed-tensors MXFP4 for GLM-5.2 and DeepSeek-V4

### PR DESCRIPTION
**Pull request by tangtang1152**
### What this PR does / why we need it?

This PR adds compressed-tensors mixed MXFP4/MXFP8 support for vLLM Ascend, including the configuration parsing and packed MXFP4 checkpoint loading required by models such as GLM-5.2 and DeepSeek-V4.

It includes three parts:

1. Support mixed-precision compressed-tensors configurations, preserve per-group activation quantization arguments, and detect MXFP8/MXFP4 quantization schemes from per-group formats.
2. Support packed MXFP4 checkpoint weight names for compressed-tensors while preserving the existing non-compressed-tensors loading path.
3. Add an LLM-Compressor mixed MXFP4/MXFP8 MoE quantization example, with attention and shared experts using MXFP8 and routed experts using MXFP4.

For compressed-tensors checkpoints, packed weights are loaded using `weight_packed`, `w13_weight_packed`, and `w2_weight_packed`, then restored to the existing runtime weight names during `process_weights_after_loading()`.

This keeps the checkpoint-format adaptation in the quantization layer without introducing model-specific branches or changing the existing runtime weight layout.

### Does this PR introduce any user-facing change?

Yes. This enables compressed-tensors mixed MXFP4/MXFP8 configurations to be recognized and MXFP4 packed checkpoint weights to be loaded by vLLM Ascend.

No new environment variable, CLI option, or user configuration item is introduced.

### How was this patch tested?

* Added unit tests for mixed-precision compressed-tensors configuration parsing and MXFP4/MXFP8 quantization type detection.
* Added unit tests for compressed-tensors packed MXFP4 Linear/MoE weight names and packed-parameter renaming.
* Verified the existing non-compressed-tensors MXFP4 weight path remains covered by the existing tests.
* The MXFP4 loading path has been validated with GLM-5.2 and DeepSeek-V4 compressed-tensors checkpoints.
* Verified the modified unit tests with Ruff and `git diff --check`; the targeted unit tests pass.

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

**Comment by tangtang1152** [comment]
Hi @kunpengW-code, the CI gate requires the `ready-precise` or `ready-all` label to run the selected tests. I don't have permission to add labels. Could you please add `ready-precise` when convenient? Thanks!

File: vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py [hunk 1]
```diff
@@ -37,6 +37,16 @@
...
 
 
+# Select checkpoint parameter names for packed compressed-tensors weights.
+def _use_packed_mxfp4_weight_names() -> bool:
+    quant_config = get_current_vllm_config().quant_config
+    if quant_config is None:
+        return False
+
+    get_name = getattr(quant_config, "get_name", None)
+    return callable(get_name) and get_name() == "compressed-tensors"
```
Comment (kunpengW-code) [inline vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py]: The hardcoded `"compressed-tensors"` here can be replaced with `COMPRESSED_TENSORS_METHOD` in vllm_ascend/utils.py

File: vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py [hunk 2]
```diff
@@ -217,6 +236,20 @@ def get_eplb_weight_views(layer: torch.nn.Module) -> list[torch.Tensor]:
...
+                requires_grad=False,
+            )
+            delattr(layer, "w13_weight_packed")
+
+        if hasattr(layer, "w2_weight_packed"):
+            layer.w2_weight = torch.nn.Parameter(
+                layer.w2_weight_packed.data,
+                requires_grad=False,
+            )
+            delattr(layer, "w2_weight_packed")
```
Comment (kunpengW-code) [inline vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py]: The logic of these code segments for reassigning values is consistent. Extracting them into a common function can reduce code duplication.

**Comment by kunpengW-code** [comment]
You can refer to other quantization example scripts in the `examples/quantization/llm-compressor` directory and add the quantization script corresponding to the new quantization mode in this PR.

**Review by kunpengW-code — DISMISSED** [review]

File: vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py [hunk 3]
```diff
@@ -37,6 +38,35 @@
 from ..registry import register_scheme
 
 
+# Select checkpoint parameter names for packed compressed-tensors weights.
+def _use_packed_mxfp4_weight_names() -> bool:
```
Comment (lcfenglinwan) [inline vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py]: It is recommended to modify the name here, for example, `_use_compressed_tensors_mxfp4_weight_names`, because this function appears to be used only by compressed-tensors.

File: vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py [hunk 4]
```diff
@@ -62,7 +92,8 @@ def __init__(self):
         self.group_size = vllm_config.quant_config.quant_description.get("group_size", 32)
 
     def get_weight(self, input_size: int, output_size: int, params_dtype: torch.dtype) -> dict[str, Any]:
-        params_dict = {"weight": torch.empty(output_size, input_size // 2, dtype=torch.uint8)}
+        weight_name = "weight_packed" if _use_packed_mxfp4_weight_names() else "weight"
```
Comment (lcfenglinwan) [inline vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py]: Why change the name for compressed-tensors? Will this *_packed style name be used elsewhere?

**Review by yiz-liu — COMMENTED** [review]
Add test case.

File: vllm_ascend/quantization/configs/compressed_tensors_config.py [hunk 5]
```diff
@@ -375,6 +403,20 @@ def _detect_quant_type(
...
+        """Detect MXFP group-wise weight and dynamic activation quantization."""
+        if input_quant is None:
+            return False
+
+        is_float = weight_quant.type == input_quant.type == QuantizationType.FLOAT
+        is_expected_bits = weight_quant.num_bits == input_quant.num_bits == num_bits
+        is_group = weight_quant.strategy == QuantizationStrategy.GROUP.value and input_quant.strategy == QuantizationStrategy.GROUP.value
+        is_dynamic = not weight_quant.dynamic and input_quant.dynamic
+        is_symmetric = weight_quant.symmetric and input_quant.symmetric
+        is_group_size_32 = weight_quant.group_size == input_quant.group_size == 32
```
[the remark this answers is not in the document]
Comment (tangtang1152) [reply vllm_ascend/quantization/configs/compressed_tensors_config.py:399]: Thanks for pointing this out. I checked the compressed-tensors `QuantizationArgs` validation: when the strategy is `GROUP`, `group_size` is required to be a positive value, and the MXFP4/MXFP8 presets explicitly set it to 32. The config is also validated through `QuantizationArgs.model_validate()` before `_is_dynamic_group_mx()` is reached.

Therefore, an omitted `group_size` is not a valid MXFP group configuration in the current compressed-tensors schema. I plan to keep the explicit `== 32` check here to avoid treating an incomplete configuration as valid MXFP.

File: vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py (hunk 3 above)
Comment (tangtang1152) [reply vllm_ascend/quantization/methods/w4a4/w4a4_mxfp4.py]: Thanks, renamed the helper to `_use_compressed_tensors_mxfp4_weight_names` to make its compressed-tensors-specific scope explicit.

**Comment by kunpengW-code** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by tangtang1152** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by kunpengW-code — APPROVED** [review]

**Comment by tangtang1152** [comment]
vamos！
